cpu/sam0_common/periph/rtc.c

```diff
@@ -94,22 +133,32 @@ void rtc_init(void)
     OSC32KCTRL->RTCCTRL.reg = OSC32KCTRL_RTCCTRL_RTCSEL_ULP1K;
 
 #endif /* ULTRA_LOW_POWER_INTERNAL_OSC_SOURCE */
+}
+#endif /* CPU_SAMD21 - Clock Setup */
 
-    /* Software Reset the RTC */
-    RTC->MODE2.CTRLA.bit.SWRST = 1;
-    /* Wait end of reset */
-    while (RTC->MODE2.CTRLA.bit.SWRST);
+void rtc_init(void)
+{
+    rtc_poweron();
```

Comment:
I don't think it's useful to enable it right know as we will disable it a few line after.
I would say to power_off at the beginning of rtc_init and power_on at the end.

Reply:
Well you mentioned it is necessary for the device to be powered for `rtc_clock_setup()` to work on saml10.

Nothing should have enabled the RTC before this function is called, so it would stall again.

Reply:
> Well you mentioned it is necessary for the device to be powered for rtc_clock_setup() to work on saml10.

I said to call `rtc_clock_setup()` before `rtc_poweroff().`
In fact, the current code still hangs for SAML10 because we're stuck in `wait_syncbusy`. Enable or disable RTC has no effect because RTC IP is not feed by the required clock so the syncbusy bit will not changed. Thus, `rtc_poweron()` is not needed here and we should setup the clock in priority. I'm sorry if I wasn't clear enough I hope my explanations get better here.
In fact, we should Setup Clock  -> Disable RTC module (because of Enabled Protected regs) -> Init RTC -> Enable RTC module

Reply:
Ok, let's try it again then. I think having the poweron/poweroff functions wrap the ENABLE bit is a bit confusing, I've wrapped them in a separate function now and made the poweron/poweroff functions actually disable the peripheral in PM/MCLK like it is done in the RTT driver.